### PR TITLE
Add quantized BatchNorm1d module

### PR DESCRIPTION
**Pull request by mattpitkin**
Fixes #147112.

### CODE REVIEW

**Comment by janeyx99** [comment]
@jerryzh168 should changes like this be considered and implemented in pytorch/ao repo?

File: torch/ao/nn/quantized/modules/batchnorm.py [hunk 1]
```diff
@@ -48,6 +48,47 @@ def from_reference(cls, bn, output_scale, output_zero_point):
         return qbn
 
 
+class BatchNorm1d(_BatchNorm):
+    r"""This is the quantized version of :class:`~torch.nn.BatchNorm1d`."""
+
+    _NNI_BN_RELU_MODULE = nni.BNReLU2d
```
Comment (jerryzh168) [inline torch/ao/nn/quantized/modules/batchnorm.py:54]: BNReLU1d?

File: torch/ao/nn/quantized/modules/batchnorm.py [hunk 2]
```diff
@@ -48,6 +48,47 @@ def from_reference(cls, bn, output_scale, output_zero_point):
...
+
+    @classmethod
+    def from_float(cls, mod, use_precomputed_fake_quant=False):
+        return _BatchNorm.from_float(
+            cls, mod, use_precomputed_fake_quant=use_precomputed_fake_quant
+        )
+
+
 class BatchNorm2d(_BatchNorm):
     r"""This is the quantized version of :class:`~torch.nn.BatchNorm2d`."""
```
Comment (jerryzh168) [inline torch/ao/nn/quantized/modules/batchnorm.py:94]: are you enabling this for eager mode quantization? probably add an entry to https://github.com/pytorch/pytorch/blob/ae351d4d0ee0676b81f58170595d016d40cd223f/torch/ao/quantization/quantization_mappings.py#L79 and test the eager mode flow for bn1d as well

File: torch/ao/nn/quantized/modules/batchnorm.py (hunk 1 above)
Comment (mattpitkin) [reply torch/ao/nn/quantized/modules/batchnorm.py]: Thanks for spotting that. It's fixed now.

File: torch/ao/nn/quantized/modules/batchnorm.py (hunk 2 above)
Comment (mattpitkin) [reply torch/ao/nn/quantized/modules/batchnorm.py:94]: Thanks, I've added the mapping. Where might the test go? In https://github.com/pytorch/pytorch/blob/main/test/quantization/eager/test_quantize_eager_qat.py?

Comment (mattpitkin) [reply torch/ao/nn/quantized/modules/batchnorm.py:94]: @jerryzh168 is there anything more I should do for this?

**Comment by mattpitkin** [comment]
@janeyx99 would you be able to remove the stale label from this?

**Comment by mattpitkin** [comment]
@janeyx99 @jerryzh168 do you think there will be any more action on this PR?

**Comment by mattpitkin** [comment]
@janeyx99 @jerryzh168 Do you think this PR is likely to be implemented, or should I consider it to be dead?

**Comment by jerryzh168** [comment]
yeah the quantized batch norm module is no longer needed in the new pt2e quant stack
